### PR TITLE
OGM-1915 Add extra small BpkPriceSize to BpkPriceImpl

### PR DESCRIPTION
**Pull request by henrik-sky**
Extra small BpkPrice has 16 sp main text and -2dp between the main text and the trailing text.
[Figma link](https://www.figma.com/file/yN0hFyZlKL0Jwbpi0rEKYT/Backpack-Beta?type=design&node-id=16700-53812&mode=dev)

Updated story, tests, README.md

### Screenshots - Extra Small

| Day                                                                                                                                                               | Night                                                                                                                                                                            |
|-------------------------------------------------------------------------------------------------------------------------------------------------------------------|----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| extra-small | extra-small_dm |

+ [x] Check this if you have read and followed the [contributing guidelines](https://github.com/Skyscanner/backpack-android/blob/main/CONTRIBUTING.md)

Remember to include the following changes:
+ [x] Component `README.md`
+ [x] Tests

If you are curious about how we review, please read through the [code review guidelines](https://github.com/Skyscanner/backpack/blob/main/CODE_REVIEW_GUIDELINES.md)

### CODE REVIEW

File: backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/price/internal/BpkPriceImpl.kt [hunk 1]
```diff
@@ -46,7 +49,11 @@ internal fun BpkPriceAlignStart(
                 BpkText(
                     text = it,
                     color = BpkTheme.colors.textSecondary,
-                    style = if (size == BpkPriceSize.Large) BpkTheme.typography.footnote else BpkTheme.typography.caption,
+                    style = when (size) {
+                        BpkPriceSize.Large -> BpkTheme.typography.footnote
+                        BpkPriceSize.Small -> BpkTheme.typography.caption
+                        BpkPriceSize.ExtraSmall -> BpkTheme.typography.caption
```
Comment (ernestkamara) [inline backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/price/internal/BpkPriceImpl.kt:55]: nit: is this change necessary?

Comment (marianeum) [reply backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/price/internal/BpkPriceImpl.kt:55]: I think it's good to use a when statement here, however maybe you can simplify it by having the small & extrasmall in the same case. Also as the same style is used across different variations of this could you extract it to a separate (private) function/property & reuse across all of them?

File: backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/price/internal/BpkPriceImpl.kt [hunk 2]
```diff
@@ -66,7 +77,11 @@ internal fun BpkPriceAlignStart(
                 modifier = Modifier.alignByBaseline(),
                 text = price,
                 color = BpkTheme.colors.textPrimary,
-                style = if (size == BpkPriceSize.Large) BpkTheme.typography.heading2 else BpkTheme.typography.heading4,
+                style = when (size) {
```
Comment (marianeum) [inline backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/price/internal/BpkPriceImpl.kt:80]: same for this case - can you extract & reuse please?

**Review by marianeum — COMMENTED** [review]
nice changes! mainly some comments around extracting some props for reuse (not your code, but becoming more obvious that it's needed with the extra style)
